dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowEventLooper.java

```diff
@@ -74,6 +76,10 @@ public void run() {
         while (!ServerLifeCycleManager.isStopped()) {
             try {
                 workflowEvent = workflowEventQueue.poolEvent();
+                if (!ServerLifeCycleManager.isRunning()) {
+                    logger.info("The current server is not running, will drop this event: {}", workflowEvent);
+                    continue;
```

Comment:
It's better to add a sleep when waiting, otherwise, there will be an empty loop all the time leading to high CPU.

Reply:
Done